app/src/main/java/com/github/sdp_begreen/begreen/viewModels/ContestCreationViewModel.kt

```diff
@@ -0,0 +1,246 @@
+        mutableStartDate.value = date
+        return true
+    }
+
+    fun editEndDate(date: Long?): Boolean {
+        if (date == null) return false
+        if (date < System.currentTimeMillis()) return false
+        mutableEndDate.value = date
+        return true
+    }
+
+    fun editStartHour(hour: Int?): Boolean {
+        if (hour == null) return false
+        if (hour < 0 || hour > 23) return false
+        mutableStartHour.value = hour
+        return true
+    }
+
+    fun editStartMinute(minute: Int?): Boolean {
+        if (minute == null) return false
+        if (minute < 0 || minute > 59) return false
+        mutableStartMinute.value = minute
+        return true
+    }
+
+    fun editEndHour(hour: Int?): Boolean {
+        if (hour == null) return false
+        if (hour < 0 || hour > 23) return false
+        mutableEndHour.value = hour
+        return true
+    }
+
+    fun editEndMinute(minute: Int?): Boolean {
+        if (minute == null) return false
+        if (minute < 0 || minute > 59) return false
+        mutableEndMinute.value = minute
+        return true
+    }
+
+    /**
+     * Function to call to edit city flow
+     */
+    fun editCity(city: String?): Boolean {
+        if (city == null) return false
+        if (city.isEmpty()) return false
+        if (city.contains(Regex("[0-9]"))) return false
+        mutableCity.value = city
+        changeLatLongIfCorrectFromManualInput()
+        return true
+    }
+
+    /**
+     * Function to call to edit country flow
+     */
+    fun editCountry(country: String?): Boolean {
+        if (country == null) return false
+        if (country.isEmpty()) return false
+        if (country.contains(Regex("[0-9]"))) return false
+        mutableCountry.value = country
+        changeLatLongIfCorrectFromManualInput()
+        return true
+    }
+
+    /**
+     * Function to call to edit postal code flow
+     */
+    fun editPostalCode(postalCode: String?): Boolean {
+        if (postalCode == null) return false
+        if (postalCode.isEmpty()) return false
+        if (postalCode.contains(Regex("[a-zA-Z]"))) return false
+        mutablePostalCode.value = postalCode
+        changeLatLongIfCorrectFromManualInput()
+        return true
+    }
+
+    /**
+     * Function to call to edit radius flow
+     */
+    fun editRadius(radius: Double?): Boolean {
+        if (radius == null) return false
+        if (radius < 0) return false
+        mutableRadius.value = radius
+        return true
+    }
+
+    /**
+     * Function to call to edit LongLat flow
+     */
+    fun editLongLat(longLat: CustomLatLng?): Boolean {
+        if (longLat == null) return false
+        if (longLat.latitude == null) return false
+        if (longLat.latitude!! < -90 || longLat.latitude!! > 90) return false
+        if (longLat.longitude == null) return false
+        if (longLat.longitude!! < -180 || longLat.longitude!! > 180) return false
```

Comment:
We could replace those line with a single if with many conditions, what do you think? (and same for the function below if you dont mind)

Reply:
I personally hate doing one big if for those but if you insist I can make it more ugly

Reply:
That's a personal choice I guess, you can let it like this, it's perfectly fine 😃